WebContent/WEB-INF/jsp/dataSourceEdit/editMqtt.jsp

```diff
@@ -35,7 +35,13 @@
         dataSourceToSave.cleanSession=$get("cleanSession");
         dataSourceToSave.brokerMode=$get("brokerMode");
 
-        DataSourceEditDwr.saveMqttDataSource(dataSourceToSave, saveDataSourceCB);
+        if(!isValid(dataSourceToSave.updateAttempts)) {
+            messages = [createValidationMessage("updateAttempts","<fmt:message key="badIntegerFormat"/>")];
```

Comment:
1. In this way, we overwrite the global variable messages, but we need to create a local variable in the scope of the function, you can use var (the scope of the function) or preferably let (the scope of the code block), which behaves similarly to a variable in Java in the scope of the method.
2. Extract the created message into a separate variable:
`let message = createValidationMessage("updateAttempts","<fmt:message key="badIntegerFormat"/>");`
...
then:
`showDwrMessages([message]);`
or:
`let messages = [message];`
`showDwrMessages(messages);`